src/firejail/main.c

```diff
@@ -256,15 +260,38 @@ static void init_cfg(int argc, char **argv) {
 		errExit("strdup");
 
 	// build home directory name
-	cfg.homedir = NULL;
-	if (pw->pw_dir != NULL) {
-		cfg.homedir = clean_pathname(pw->pw_dir);
-		assert(cfg.homedir);
-	}
-	else {
+	if (pw->pw_dir == NULL) {
 		fprintf(stderr, "Error: user %s doesn't have a user directory assigned\n", cfg.username);
 		exit(1);
 	}
+	// resolve symbolic links
+	cfg.homedir = realpath(pw->pw_dir, NULL);
```

Comment:
Everything else is decoration, but this realpath() adds real attack surface. As a mitigation I enforced a user owned home directory, and disallowed /sys and /proc, as we don't want to end up with a writable tmpfs mounted in there. I'd very much appreciate thoughts if this would be considered "safe enough".